### PR TITLE
use cached pyenv to get python 3.8.1 for tool envs on travis

### PR DESCRIPTION
**Pull request by altendky**

### CODE REVIEW

**Comment by altendky** [comment]
Kick CI to test now that caches have been written.

**Comment by altendky** [comment]
Yeah, the extra 30 seconds of run-time isn't cool.  It looks like a bit shy of half is involved in writing the hopefully unchanged cache back.  So I'll start by seeing if I can avoid that.

**Comment by altendky** [comment]
Or I misread that and it's only an _extra_ five seconds on the cache.

**Comment by altendky** [comment]
1:00 https://travis-ci.com/python-desert/desert/jobs/280279996
1:31 https://travis-ci.com/python-desert/desert/jobs/280598075

11.4 for `travis_apt_get_update` (which is only needed if building the cache `:|`)
0.4 + 4.0 extra for pyenv cache read
2.4 extra for pip cache read
4.6 for extra cache write (why 8.5s for pip cache every time?)

I'm not sure where the other 8 or so extra seconds come from.

**Comment by python-desert** [comment]
I'd guess there's a fair amount of variation that depends on travis platform congestion etc so trying a bunch of times might be warranted if you want to go down that road. 

There's also other options involving docker/nix combinations.

**Comment by altendky** [comment]
Quite so, it's been significantly in the 1:30+ range though I didn't retrigger a master-alike build yet.  Trying master as a PR over in #60.  Looks like 1:05 - 1:10 so not quite as bad maybe but still questionable (based on this terrible comparison).

How concerning is this time increase?  I don't like it but could be talked into dealing with it later if it isn't too annoying.

**Comment by python-desert** [comment]
We can deal with it later if you like.

**Comment by altendky** [comment]
Alrighty, if you're ok with it, let's get back to adding PyPy3 testing.

File: .travis.yml [hunk 1]
```diff
@@ -6,11 +6,38 @@ branches:
     # 'v' followed by a digit, such as for version tags.
     - master
     - /^v\d.*$/
-cache: pip
+cache:
```
Comment (python-desert) [inline .travis.yml:9]: What is all this stuff? Good to have a comment explaining why they're needed.

**Review by ghost — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (python-desert) [reply .travis.yml:9]: Oops I meant to link this https://github.com/python-desert/desert/pull/59/commits/6e49644df70dce5366f81c663bda710138b74f71#diff-354f30a63fb0907d4ad57269548329e3R18-R35

Where does this list come from?

**Review by ghost — APPROVED** [review]

**Pull request merged**